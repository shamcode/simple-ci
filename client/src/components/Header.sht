{% import path from 'sham-ui-router/params' %}

<header>
    <a :hrefto={{path("project-list")}} class="logo">
        Simple CI
    </a>
</header>
