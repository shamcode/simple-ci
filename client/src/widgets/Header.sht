<header>
    <h1>client</h1>
</header>
