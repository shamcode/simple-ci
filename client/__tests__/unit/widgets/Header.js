import { DI } from 'sham-ui';
import Header from '../../../src/widgets/Header.sht';
import renderer from 'sham-ui-test-helpers';

afterEach( () => {
    DI.bind( 'router', null );
} );

it( 'renders correctly', () => {
    DI.bind( 'router', {
        generate: jest.fn().mockReturnValueOnce( '/' )
    } );

    const meta = renderer( Header, {} );
    expect( meta.toJSON() ).toMatchSnapshot();
} );
