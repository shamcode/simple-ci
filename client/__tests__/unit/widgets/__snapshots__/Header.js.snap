// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Object {
  "ID": "widget",
  "constructorOptions": Object {},
  "html": "<header><a href=\\"/\\" class=\\"logo\\">Simple CI</a></header>",
  "nodes": Array [
    Object {
      "ID": undefined,
      "constructorOptions": null,
      "html": null,
      "nodes": null,
      "options": null,
    },
  ],
  "options": Object {
    "actionSequence": Array [
      "render",
      "bindEvents",
    ],
    "renderAsync": false,
    "renderDependence": Array [],
    "types": Array [],
  },
}
`;
